### PR TITLE
Updating ad placement logic per length for gallery overlay

### PR DESCRIPTION
**Pull request by tpitre**
https://jira.wnyc.org/browse/DS-455

### CODE REVIEW

File: source/_patterns/03-organisms/gallery/gallery-overlay.json [hunk 1]
```diff
@@ -4,23 +4,22 @@
   "slides": [
     {
       "slide_title": "Mercado Little Spain",
-      "break_m": "700",
       "img": {
-        "src_s": "https://picsum.photos/400/700?image=122"
+        "src_s": "https://cmsplatypus.gothamist.com/gothamistgallery/2019/3/15/df594e893hyfood-14-jpg-mobile.jpeg"
```
Comment (walsh9) [inline source/_patterns/03-organisms/gallery/gallery-overlay.json]: just a heads up, we'll probably be turning off `cmsplatypus.gothamist.com` some time in the near future so you might want to use a different image here and elsewhere where you're pointing to this server...

File: source/_patterns/03-organisms/gallery/gallery-overlay.twig [hunk 2]
```diff
@@ -28,11 +28,20 @@
...
               {% include 'molecules-billboard' %}
             </div>
           {% endif %}
+
+          {# Ad requirements for greater than 10 total items #}
+          {% if loop.length > 10 and loop.index > 10 and loop.index is divisible by(6) %}
+            <div class="l-wrap">
+              {% include 'molecules-billboard' %}
+            </div>
+          {% endif %}
```
Comment (walsh9) [inline source/_patterns/03-organisms/gallery/gallery-overlay.twig]: I discussed the ad cadence with Kate and I think we've managed to simplify the description of the ad conditions. Basically it's "show ads after every 6th slide, and also show one after the 3rd slide". No more different conditions based on the total number of ads. I think we can probably simplify the code with this logic too.

I'm also going to update this on the tickets.

e.g.
```
1
2
3 (three)
***AD***
4
5
6 (multiple of 6)
***AD***
7
8
9
10
11
12 (multiple of 6)
***AD***
13
14
15
16
17
18 (multiple of 6)
***AD***
19
20
21
22
23
24 (multiple of 6)
***AD***
25
...
```

File: source/scss/_library/_objects.text.scss [hunk 3]
```diff
@@ -506,6 +506,12 @@ a.o-breadcrumb {
     position: absolute;
   }
 
+  .u-path-fill--white {
+    path {
```
Comment (walsh9) [inline source/scss/_library/_objects.text.scss]: If we want this rule to support a wider range of svgs, we might consider a broader set of selectors here. We might not need this now though. I don't know enough about where this is used to know if this really matters. Up to you.
```
circle,ellipse,line,path,polygon,polyline,rect /* svg shapes*/ {
```

File: source/_patterns/03-organisms/gallery/gallery-overlay.json (hunk 1 above)
Comment (tpitre) [reply source/_patterns/03-organisms/gallery/gallery-overlay.json]: Gotcha!

File: source/scss/_library/_objects.text.scss (hunk 3 above)
Comment (tpitre) [reply source/scss/_library/_objects.text.scss]: Good point. I'll just target `svg *`.

File: source/_patterns/03-organisms/gallery/gallery-overlay.twig (hunk 2 above)
Comment (tpitre) [reply source/_patterns/03-organisms/gallery/gallery-overlay.twig]: @walsh9 This is what it's doing right now - https://nypr.southleft.com/?p=pages-article-gallery-long

That extra piece of logic is to break out of the "every 3" and move to an "every 6".

File: source/scss/_library/_objects.text.scss (hunk 3 above)
Comment (tpitre) [reply source/scss/_library/_objects.text.scss]: This has been updated.

File: source/_patterns/03-organisms/gallery/gallery-overlay.twig (hunk 2 above)
Comment (walsh9) [reply source/_patterns/03-organisms/gallery/gallery-overlay.twig]: i was just thinking you could simplify to a single block. something like:
```
{% if loop.index is divisible by(6) or loop.index = 3 %}
```

Comment (tpitre) [reply source/_patterns/03-organisms/gallery/gallery-overlay.twig]: Ah, I see. That's much cleaner. Updated!

**Pull request merged**